### PR TITLE
Add Habana to Azure pipes to support HPU Tests

### PR DESCRIPTION
**Pull request by raoakarsha**
## What does this PR do?

Fixes #\<issue_number>

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: .azure-pipelines/hpu-tests.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    include:
+      - "master"
+      - "release/*"
+      - "refs/tags/*"
+pr:
+  - "master"
+  - "release/*"
+
+jobs:
+  - job: HpuTests
```
Comment (Borda) [inline .azure-pipelines/hpu-tests.yml:17]:
```suggestion
  - job: Hpu
```

File: .azure-pipelines/hpu-tests.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    pool: intel-hpus
+
+    workspace:
+      clean: all
+
+    steps:
+
+    - bash: |
+        echo "Installing pytorch_lightning and required dependencies"
+        python setup.py bdist_wheel
```
Comment (Borda) [inline .azure-pipelines/hpu-tests.yml:33]:
```suggestion
```
this is not needed

File: .azure-pipelines/hpu-tests.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        pip install -U ./dist/pytorch_lightning*
+        pip install . --requirement requirements/hpu.txt
+      displayName: 'Install dependencies'
+      continueOnError: true
+
+    - bash: |
+         echo "Run Hpu tests in parallel"
+         python ".azure-pipelines/run_hpu_tests.py"
```
Comment (Borda) [inline .azure-pipelines/hpu-tests.yml:41]:
```suggestion
         python .azure-pipelines/run_hpu_tests.py
```

File: .azure-pipelines/hpu-tests.yml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    - bash: |
+        echo "Installing pytorch_lightning and required dependencies"
+        python setup.py bdist_wheel
+        pip install -U ./dist/pytorch_lightning*
+        pip install . --requirement requirements/hpu.txt
+      displayName: 'Install dependencies'
+      continueOnError: true
+
+    - bash: |
+         echo "Run Hpu tests in parallel"
```
Comment (Borda) [inline .azure-pipelines/hpu-tests.yml:40]:
```suggestion
```

File: .azure-pipelines/hpu-tests.yml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+    pool: intel-hpus
+
+    workspace:
+      clean: all
+
+    steps:
+
+    - bash: |
+        echo "Installing pytorch_lightning and required dependencies"
```
Comment (Borda) [inline .azure-pipelines/hpu-tests.yml:32]:
```suggestion
```
you already have the step names

File: .azure-pipelines/hpu-tests.yml [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+      continueOnError: true
+
+    - bash: |
+         echo "Run Hpu tests in parallel"
+         python ".azure-pipelines/run_hpu_tests.py"
+      displayName: 'HPU Tests in parallel'
+      continueOnError: true
+
+    - bash: |
+        echo "Coverage report:"
```
Comment (Borda) [inline .azure-pipelines/hpu-tests.yml:46]:
```suggestion
```

File: .azure-pipelines/hpu-tests.yml [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+        python -m coverage report
+        python -m coverage xml
+        python -m coverage html
+        python -m codecov --token=$(CODECOV_TOKEN) --commit=$(Build.SourceVersion) --flags=pytest --name="HPU Test Coverage"
```
Comment (Borda) [inline .azure-pipelines/hpu-tests.yml:50]:
```suggestion
        python -m codecov --token=$(CODECOV_TOKEN) --commit=$(Build.SourceVersion) --flags=pytest,hpu --name="HPU Test Coverage"
```

File: .azure-pipelines/run_hpu_tests.py [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+
+    Args:
+        exit_status: The returned exit_status after executing run_hpu_tests_parallel().
+    Return of this function will be the return to main().
+    Based on the exit status of the HPU tests, we return success or failure to the main method.
+    """
+    if all(v == 0 for v in exit_status):
+        print("All HPU tests passed")
+        file_name = "stdout_log.txt"
+        print_logs(file_name)
```
Comment (Borda) [inline .azure-pipelines/run_hpu_tests.py:121]: why not one print?

File: .azure-pipelines/hpu-tests.yml (hunk 3 above)
Comment (Borda) [reply .azure-pipelines/hpu-tests.yml:41]: Also, this does not collect any coverage so following reporting is redundant

File: .azure-pipelines/hpu-tests.yml (hunk 1 above)
Comment (raoakarsha) [reply .azure-pipelines/hpu-tests.yml:17]: done

File: .azure-pipelines/hpu-tests.yml (hunk 5 above)
Comment (raoakarsha) [reply .azure-pipelines/hpu-tests.yml:32]: removed echo in all the stages as we have the step names defined

File: .azure-pipelines/hpu-tests.yml (hunk 2 above)
Comment (raoakarsha) [reply .azure-pipelines/hpu-tests.yml:33]: we need this to install pytorch lightning wheel

File: .azure-pipelines/hpu-tests.yml (hunk 7 above)
Comment (raoakarsha) [reply .azure-pipelines/hpu-tests.yml:50]: done

File: .azure-pipelines/run_hpu_tests.py (hunk 8 above)
Comment (raoakarsha) [reply .azure-pipelines/run_hpu_tests.py:121]: we do have two files while running the code in sub process, only when we have the error exit status we display the error_log else we only print the stdout_log

Comment (raoakarsha) [reply .azure-pipelines/run_hpu_tests.py:121]: Also print_logs is the method name

File: .azure-pipelines/hpu-tests.yml [hunk 9]
```diff
@@ -30,3 +30,18 @@ jobs:
     - bash: |
         hwinfo --short
       displayName: 'Instance HW info'
+
+    - bash: |
+        pip install . --requirement requirements/hpu.txt
```
Comment (Borda) [inline .azure-pipelines/hpu-tests.yml:35]:
```suggestion
        pip install . --requirement requirements/test.txt
```

**Comment by Borda** [comment]
@raoakarsha seems that this PR does hot include the integration so the testing will fail always...
 so I would suggest merging this PR to the #11808

**Review by Borda — CHANGES_REQUESTED** [review]
@raoakarsha seems that this PR does hot include the integration so the testing will fail always...
so I would suggest merging this PR to the https://github.com/PyTorchLightning/pytorch-lightning/pull/11808

**Pull request closed without merging**